bin/find-device-port-macos

```diff
@@ -52,32 +54,33 @@ sub try_for_raw_serialnum {
     # filename. I'm not sure why, but system_profiler has a serial number ending in "E",
     # whereas the device filename ends in "1". In fact, when I change HID.getShortName()
     # to return "kbio02", the final character is replaced with a "1", so we should do the
-    # same here. If the serial number doesn't end in a digit, however, we may want to append
-    # rather than replace the last character with a 1.
-    if ($serial_port_name =~ /\d$/) {
-        $serial_port_name =~ s/.$/1/;
-    } else {
-        $serial_port_name .= "1";
-    }
+    # same here. 
+    exit_with_port_if_exists($serial_port_name . "1");
+    
+    # now try with the last character chop-ped off 
+    chop $serial_port_name;
     exit_with_port_if_exists($serial_port_name);
+    exit_with_port_if_exists($serial_port_name . "1");
 }
 
 sub try_for_location_id {
     my $location_id = shift;
 
     # Here, also, the final character is always a "1", so if macOS ever stops doing that, this
-    # will need an update, as well.
-    my $loc = substr($location_id, 2, 3);
-    exit_with_port_if_exists("/dev/cu.usbmodem" . $loc . 1);
+    # will need an update, as well. 
+    if ($location_id =~ /0x(\d+?)0*\b/) { 
+        exit_with_port_if_exists("/dev/cu.usbmodem" . $1 . 1);
```

Comment:
using $1 directly there can be a little bit brittle and not assigning it to a variable misses an opportunity for a descriptive name to help make the code more readable.

What problem was this change fixing?

Reply:
The location string is not necessarily 3 chars long.